backend/module/chaos/serverexperimentation/xds/ecds_faults_generator.go

```diff
@@ -172,69 +114,57 @@ func generateECDSResource(experiments []*experimentstore.Experiment, cluster str
 		}
 	}
 
-	logger.Debugw("ECDS Fault filter", "faultFilter", faultFilter, "upstreamcluster", upstreamCluster)
-
 	serializedFaultFilter, err := proto.Marshal(faultFilter)
 	if err != nil {
-		logger.Warnw("Unable to unmarshal fault filter", "faultFilter", faultFilter)
-		return []gcpTypes.ResourceWithTtl{}
+		return nil, err
 	}
 
-	return []gcpTypes.ResourceWithTtl{{
-		Resource: &gcpCoreV3.TypedExtensionConfig{
-			Name: faultFilterName,
-			TypedConfig: &any.Any{
-				TypeUrl: faultFilterTypeURL,
-				Value:   serializedFaultFilter,
-			},
+	config := &gcpCoreV3.TypedExtensionConfig{
+		Name: faultFilterName,
+		TypedConfig: &any.Any{
+			TypeUrl: faultFilterTypeURL,
+			Value:   serializedFaultFilter,
 		},
-		Ttl: ttl,
-	}}
+	}
+
+	return xds.NewECDSResource(enforcingCluster, config)
 }
 
-func generateEmptyECDSResource(cluster string, ecdsConfig *ECDSConfig, logger *zap.SugaredLogger) []gcpTypes.ResourceWithTtl {
-	// cache serialized
-	if SerializedDefaultFaultFilter == nil {
+func (g *ECDSFaultsGenerator) GenerateDefaultResource(cluster string, resourceName string) (*xds.ECDSResource, error) {
+	if resourceName != faultFilterConfigNameForIngressFault && !strings.HasPrefix(resourceName, fmt.Sprintf(faultFilterConfigNameForEgressFault, "")) {
+		return xds.NewEmptyECDSResource(), nil
```

Comment:
i wonder if it would be nicer if we could just return nil and the generator would do the right thing

Reply:
Does returning nil, nil feels idiomatic in Go? Feels error prone (accessing nils -> crashing)

Reply:
Pretty common: https://github.com/search?q=%22return+nil%2C+nil%22&type=code

As long as the contract specifies that nil is permitted it seems fine to me. The type signature still allows nil, so without nil handling a person trying to use this might run into nil defererences anyways

Reply:
most of these are for cases where a method returns `nil, nil, err` or forks of eth crypto client. Anyway, I can make a change - I think it's a reasonable ask.